src/ring_theory/localization.lean

```diff
@@ -891,54 +892,84 @@ lemma map_mem_non_zero_divisors {B : Type*} [integral_domain B] {g : A →+* B}
 λ z hz, eq_zero_of_ne_zero_of_mul_eq_zero
   (map_ne_zero_of_mem_non_zero_divisors hg) hz
 
-variables (K : Type*)
+variables {K}
 
-/-- Localization map from an integral domain `R` to its field of fractions. -/
-@[reducible] def fraction_map [comm_ring K] := localization_map (non_zero_divisors R) K
+/-- Submonoids in which localization of an integral domain is an integral domain
+TODO: should this have a class annotation or is that not useful? -/
+@[class]
+def le_non_zero_divisors [comm_ring K] (M : submonoid K) := M ≤ non_zero_divisors K
```

Comment:
Thanks for this PR! It looks good, and I only have comments (as you anticipated :rofl:) about this def. Making this a class feels to me like it will be abusing the type class system to prove things that it isn't really made for. So I wouldn't really make this a class.
And once it is no longer a class, you might as well inline the definition where you need it, and remove this definition completely.
What do you think?

Reply:
I think its reasonable to just inline this, the only question would be if I should create versions of these lemmas that take no inputs in the case that the localization map is `fraction_map`, or if I should change all the current uses of these theorems to pass in `le_of_eq rfl` as the proof of this every time they call one of these?

Reply:
The special versions for fraction map are a nice gesture to future users (-;
And the proofs are trivial, so if you want to add them: yes please :smiley:

Reply:
I went and added them as part of removing the class definition, which has the upside that I don't need to refactor any of the existing uses of these theorems.